src/components/new-request/name-input.vue

```diff
@@ -34,38 +29,43 @@ import { MRAS_MAX_LENGTH } from '@/components/new-request/constants'
 
 @Component({})
 export default class NameInput extends Vue {
-  // Refs
-  $refs!: {
-    nameInput: any
-  }
+  /** Whether to perform MRAS search. */
+  @Prop({ default: false }) readonly isMrasSearch!: boolean
+  /** Whether this component is read-only (eg, on name check page). */
+  @Prop({ default: false }) readonly isReadOnly!: boolean
+  /** Hint to show (eg, on name check page). */
+  @Prop({ default: null }) readonly hint!: string
 
-  // Global getters
+  // Store getters
   @Getter getCorpSearch!: string
   @Getter getErrors!: string[]
+  @Getter getHasNoCorpNum!: boolean
+  @Getter getIsXproFlow!: boolean
   @Getter getLocation!: Location
   @Getter getName!: string
   @Getter getRequestActionCd!: NrRequestActionCodes
-  @Getter getIsXproMras!: boolean
+  @Getter isMrasJurisdiction!: boolean
 
-  // Global actions
-  @Action setClearErrors!: ActionBindingIF
+  // Store actions
+  @Action setClearErrors!: () => void
   @Action setCorpSearch!: ActionBindingIF
   @Action setName!: ActionBindingIF
   @Action setMrasSearchInfoModalVisible!: ActionBindingIF
   @Action startAnalyzeName!: ActionBindingIF
 
-  // Props
-  @Prop({ default: false }) readonly isSearchAgain!: boolean
-  @Prop({ default: false }) readonly isMrasSearch!: boolean
-  @Prop({ default: false }) readonly isReadOnly!: boolean
-  @Prop({ default: null }) readonly hint!: string
-
-  private err_msg = 'Cannot exceed ' + MRAS_MAX_LENGTH + ' characters'
+  readonly err_msg = 'Cannot exceed ' + MRAS_MAX_LENGTH + ' characters'
 
-  additionalRules = [
+  readonly defaultRules = [
     v => (!v || v.length <= MRAS_MAX_LENGTH) || this.err_msg
   ]
 
+  nameInputComponent = null
+
+  mounted (): void {
+    // ref is only valid after component is mounted
+    this.nameInputComponent = this.$refs['nameInputRef']
```

Comment:
I needed to do this or the computed `isCorpValid` threw an error before this component was mounted and was not reactive after. I believe this was an existing issue.